Source/Urho3D/IO/PackageBuilder.h

```diff
@@ -0,0 +1,82 @@
+//
+// Copyright (c) 2008-2020 the Urho3D project.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+#pragma once
+
+namespace Urho3D
+{
+class File;
+class MemoryBuffer;
+
+/// A helper class to create PackageFile.
+class URHO3D_API PackageBuilder
+{
+public:
+    /// Constructor.
+    PackageBuilder();
+
+    /// Build package.
+    bool Create(AbstractFile* buffer, bool compress);
+
+    /// Append entry to package.
+    bool Append(const ea::string& name, const SharedPtr<File>& file);
```

Comment:
`SharedPtr<>` should not be used in function parameters. Caller guarantees lifetime of file anyway. Also this and `MemoryBuffer` version below could be collapsed to a single function that takes `Deserializer` as a parameter.

Reply:
Deserializer  doesn't have Seek or GetPosition.

Reply:
But it does 👀